### PR TITLE
debian: dynamic distributions

### PR DESCRIPTION
**Pull request by hdonnay**
These changes bring the `debian` package in-line with the new auto discovering scheme.

### CODE REVIEW

**Comment by hdonnay** [comment]
forgot I need to improve a grip of log lines

File: updater/defaults/defaults.go [hunk 1]
```diff
@@ -56,9 +56,13 @@ func inner(ctx context.Context) error {
 		return err
 	}
 	updater.Register("ubuntu", uf)
+	df, err := debian.NewFactory(ctx)
+	if err != nil {
+		return err
+	}
+	updater.Register("debian", df)
```
Comment (crozzy) [inline updater/defaults/defaults.go:63]: considering the constructor is doing:
```go
func NewFactory(ctx context.Context) (*Factory, error) {
	f := &Factory{}
	return f, nil
}
```
do we need to return the error (and subsequently instantiate it in this different way?

File: debian/updater.go [hunk 2]
```diff
@@ -1,98 +1,357 @@
...
 
 var (
-	_ driver.Updater      = (*Updater)(nil)
-	_ driver.Configurable = (*Updater)(nil)
+	// LinkRegexp is a bad regexp to extract link targets.
+	// This will break if Debian's codenames include a double-quote in the future.
+	linkRegexp = regexp.MustCompile(`href="([^"]+)"`)
+	// SkipList is a list of strings that, experimentally, indicate the string
+	// is not a codename.
+	skipList = []string{
```
Comment (crozzy) [inline debian/updater.go:128]: Could possibly add `updates` and `backports` here to avoid more requests

**Review by crozzy — COMMENTED** [review]
LGTM, just a couple of comments

File: updater/defaults/defaults.go (hunk 1 above)
Comment (hdonnay) [reply updater/defaults/defaults.go:63]: I'm just making it look like the new, future proofed constructor functions.

**Review by crozzy — APPROVED** [review]

**Comment by hdonnay** [comment]
clean rebase, so I'm merging with superpowers.

**Pull request merged**